### DISCUSSION TITLE
DatePicker 2-digit years

### DISCUSSION THREAD

**Discussion by bk-tho** [adobe/react-spectrum#5071] [Q&A]
We use components that facilitate both `useDatePicker` and `useDateRangePicker` in various places of our application. Lately we received feedback from our customers that they often make mistakes by accident when using those components. Mainly there are 2 concerns that we hear about repeatedly:

**1. The DatePicker sometimes limits day of month to 30**: There's [an issue for this](https://github.com/adobe/react-spectrum/issues/3256) and you guys already decided to "fix" that behavior. So that's all good 👍 If anything, I'd like to promote that issue even more since I can confirm that many of our customers do indeed find this confusing and annoying.
**2. People often type in 2-digit years**: People do `23.09.23` when in fact they mean `23.09.2023`. This has lead to a number of inquiries where we needed to tell our customers that `23` means `23` and not `2023`. In our application, there is practically no use-case for specifying any date that is not in the 20xx range. 
Would it be possible to offer an option to interpret 2-digit years in such a way that it automatically fills those up with a leading `20`?  I understand that it's not generally desirable to implement such behavior but it might be a use-case common-enough to provide first-class support for it. Otherwise we'd have to work our way around it in userland.

**Comment by LFDanLu** [comment]
Thank you for chiming in with regards to https://github.com/adobe/react-spectrum/issues/3256, I'll bring this up in our next grooming session to see if we could bump it up in priority. As for your second question, I think that is a reasonable request, curious if we have a need for the same thing in Adobe products as well (if we do, we can discuss with our Spectrum design team and bump in priority). Would you mind filing an "enhancement" issue for greater visibility?

**Reply by bk-tho** [reply]
Thanks for your quick response! I filed an issue here https://github.com/adobe/react-spectrum/issues/5081
